pymc/model.py

```diff
@@ -1937,11 +1937,45 @@ def Point(*args, filter_model_vars=False, **kwargs) -> Dict[str, np.ndarray]:
 
 
 def Deterministic(name, var, model=None, dims=None, auto=False):
-    """Create a named deterministic variable
+    """Create a named deterministic variable.
+
+    Deterministic nodes are only deterministic given all of their inputs, i.e.
+    they don't add randomness to the model.  They are generally used to record
+    an intermediary result.
+
+    Indeed, PyMC allows for arbitrary combinations of random variables, for
+    example in the case of a logistic regression
+
+    .. code:: python
+
+        with pm.Model():
+            alpha = pm.Normal("alpha", 0, 1)
+            intercept = pm.Normal("intercept", 0, 1)
+            p = pm.math.invlogit(alpha * x + intercept)
+            outcome = pm.Bernoulli("outcome", p, observed=outcomes)
+
+
+    but cannot memorize the fact that the expression ``pm.math.invlogit(alpha *
```

Comment:
It can memorize but chooses not to because memory/ speed optimizations can be done when intermediate computations don't have to be returned.

Also worth mentioning that deterministics will only be computed once per sample, wheres the [d]logp graph without them may be computed 1000x times during a single NUTS step

Reply:
Thank you for the precision! I will try to improve the text.

> Also worth mentioning that deterministics will only be computed once per step, whereas the [d]logp graph without them may be computed 1000x times during a single NUTS step

I am sorry but I am not sure that I understand what you mean. Does this imply something in terms of sampling efficiency?

Reply:
I meant that we never compute Deterministics within a step (where we might evaluate the graph 1000x times), only at the end once

Reply:
Ah yes that makes sense, thank you!

Reply:
Yes!